.github/workflows/generate.yml

```diff
@@ -0,0 +1,69 @@
+# ---------------------------------------------------------------------------
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ---------------------------------------------------------------------------
+
+name: generate
+
+on:
+  pull_request:
+    branches:
+      - main
+      - "release-*"
+    paths-ignore:
+      - 'docs/**'
+      - 'proposals/**'
+      - '**.adoc'
+      - '**.md'
+      - 'KEYS'
+      - 'LICENSE'
+      - 'NOTICE'
+  push:
+    branches:
+      - main
+      - "release-*"
+    paths-ignore:
+      - 'docs/**'
+      - 'proposals/**'
+      - '**.md'
+      - '**.adoc'
+      - 'KEYS'
+      - 'LICENSE'
+      - 'NOTICE'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  generate:
+    strategy:
+      matrix:
+        # TODO: test an all the supported OS
+        # [ubuntu-20.04, macos-latest, windows-latest]
+        os: [ubuntu-20.04]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          # make generate requires Go >=1.17
+          go-version: 1.17.x
```

Comment:
We're still on 1.16. We may need to wait #2845 before merging this.

Reply:
> doesn't commit any changes to the repo

I thought it was adding the generate doc back to the repo. Can you please clarify what the action does?

Reply:
Yeah, the workflow just tests `make generate`; it makes sure the target runs fine, as I found this target is fragile and easy to be in a state of not working, e.g. it doesn't run with Go 1.16 or sometimes `go mod tidy` has tidied up too much of `go.mod` . We discover that only when we try to run the target. This workflow should find such breaking changes earlier.

Reply:
Okey, so, the idea of the action is to run it just to report any problem, not to generate on behalf of the user. I'd call it `verify-generate` to avoid creating confusion on the goal of the action.

Reply:
Thanks, the suggested name is better. In the future, it might be worth adding the 'real' generate workflow that automatically generates docs for us. But let's do it another time.